spec.bs

```diff
@@ -287,17 +287,118 @@ href="https://html.spec.whatwg.org/multipage/embedded-content-other.html#dimensi
+
+<div algorithm>
+  To <dfn for="fenced frame config mapping" export>store a pending config</dfn> into a particular
+  [=fenced frame config mapping=] given a [=fenced frame config=] |config|, run these steps:
+
+  1. Let |pendingMapping| be the [=fenced frame config mapping/pending config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If the [=map/size=] of the [=fenced frame config mapping/pending config mapping=] + the
+     size of the [=fenced frame config mapping/finalized config mapping=] meets or
+     exceeds the [=fenced frame config mapping/maximum number of configs=], return failure.
+
+  1. Let |urn| be a randomly generated [=urn uuid=].
+
+  1. [=map/Set=] |pendingMapping|[|urn|] to |config|.
+</div>
+
+<div algorithm>
+  To <dfn for="fenced frame config mapping" export>finalize a pending config</dfn> in a particular
+  [=fenced frame config mapping=] given a [=urn uuid=] |urn|, run these steps:
+
+  1. Let |pendingMapping| be the [=fenced frame config mapping/pending config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. Let |finalizedMapping| be the [=fenced frame config mapping/finalized config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If |pendingMapping|[|urn|] does not [=map/exist=], return failure.
+
+  1. Let |finalizedConfig| be |pendingMapping|[|urn|].
+
+  1. [=map/Remove=] |pendingMapping|[|urn|].
+
+  1. [=map/Set=] |finalizedMapping|[|urn|] be |finalizedConfig|.
+</div>
+
+<div algorithm>
+  To <dfn for="fenced frame config mapping" export>store nested configs</dfn> into a particular
+  [=fenced frame config mapping=] given a [=fenced frame config instance/nested configs=]
+  |nestedConfigs|, run these steps:
+
+  1. Let |nestedMapping| be the [=fenced frame config mapping/nested config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If |nestedConfigs| is null, return.
+
+  1. [=map/iterate|For each=] |urn| → |config| of |nestedConfigs|:
+
+  1. [=map/Set=] |nestedMapping|[|urn|] to |config|.
+</div>
+
+<div algorithm>
+  To <dfn for="fenced frame config mapping" export>find a config</dfn> in a particular [=fenced
+  frame config mapping=] given a [=urn uuid=] |urn|, run these steps:
+
+  1. Let |nestedMapping| be the [=fenced frame config mapping/nested config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. Let |pendingMapping| be the [=fenced frame config mapping/pending config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. Let |finalizedMapping| be the [=fenced frame config mapping/finalized config mapping=] of
+     the associated [=fenced frame config mapping=].
+
+  1. If |nestedMapping|[|urn|] [=map/exists=], return its value.
+
+  1. If |pendingMapping|[|urn|] [=map/exists=], wait until it does not [=map/exist=].
+     <span class="XXX">TODO: Formalize waiting?</span>
```

Comment:
This is the piece that's intended to be called during navigation while in parallel, waiting for a pending config right?